sklearn/manifold/tests/test_mds.py

```diff
@@ -55,10 +55,85 @@ def test_smacof_error():
         mds.smacof(sim, init=Z, n_init=1)
 
 
+def test_svd():
+    # Test svd using example data from "An Introduction to MDS"
+    # Florian Wickelmaier, p 11
+    sim = np.array([[0, 93, 82, 133],
+                    [93, 0, 52, 60],
+                    [82, 52, 0, 111],
+                    [133, 60, 111, 0]])
+
+    X, stress = mds.svd(sim, n_components=2)
+    X_true_1 = np.array([[-62.831, -32.97448],
+                         [18.403, 12.02697],
+                         [-24.960, 39.71091],
+                         [69.388, -18.76340]])
+    X_true_2 = np.copy(X_true_1)
+    X_true_2[:, 0] *= -1
+
+    # Signs of columns are dependent on signs of computed eigenvectors
+    # which are arbitrary and meaningless
+    assert(np.allclose(X, X_true_1)
+           or np.allclose(X, -X_true_1)
+           or np.allclose(X, X_true_2)
+           or np.allclose(X, -X_true_2))
+
+
+def test_svd_error():
+    # Non symmetric (dis)similarity matrix:
+    sim = np.array([[0, 5, 9, 4],
+                    [5, 0, 2, 2],
+                    [3, 2, 0, 1],
+                    [4, 2, 1, 0]])
+
+    with pytest.raises(ValueError):
```

Comment:
better to specifically check for the error message with 
```suggestion
    with pytest.raises(ValueError, match='...'):
```

Reply:
same for the other ones.

Reply:
I'm not sure of that changes, because some exceptions are raised in method 'check_symmetric', where error messages aren't specified.

Reply:
check_symmetric does seem to specify the error messages? Namely `array must be 2-dimensional and square` and `Array must be symmetric`.

I agree it's always good to use `match=`, if only to let us know what the user will see in case of an error

Reply:
I've meant specified somewhere in docs. I don't feel like making tests dependent on undocumented piece of code from another package is best practise, as another packages should be treated as blackboxes in tests, without any 'magic' insight. But maybe it doesn't matter. I'll do as you wish.

Reply:
I agree but in this case this is `check_symmetric` which is a public scikit-learn util, we have entire control over it

Reply:
Ok. Done